### PR TITLE
Untypeast: define untype_expression, untype_pattern

### PR DESCRIPTION
**Pull request by trefis**
Just exposing some convenience functions that we've been defining in merlin for the past few years.

### CODE REVIEW

File: typing/untypeast.mli [hunk 1]
```diff
@@ -83,3 +83,6 @@ val untype_structure : ?mapper:mapper -> Typedtree.structure -> structure
 val untype_signature : ?mapper:mapper -> Typedtree.signature -> signature
 
 val constant : Asttypes.constant -> Parsetree.constant
+
+val untype_expression : ?mapper:mapper -> Typedtree.expression -> expression
+val untype_pattern : ?mapper:mapper -> _ Typedtree.general_pattern -> pattern
```
Comment (gasche) [inline typing/untypeast.mli:85]: Nit: I would find it slightly more natural to see those functions listed right after the two other `untype_` and before `constant`.

**Review by gasche — APPROVED** [review]
Good to merge if CI is green.

**Comment by Drup** [comment]
(hot take that doesn't prevent merging, on the contrary: all these functions should loose the prefix `untype`)

**Comment by trefis** [comment]
> (hot take that doesn't prevent merging, on the contrary: all these functions should loose the prefix untype)

I'm going to ignore this (for now, and indefinitely): compiler-libs API is what it is, I do not have strong enough shoulders to start cleaning it up and dealing with people complaining about the resulting breakage.

But if you feel brave enough, rest assured that I will not stand in your way.

**Pull request merged**